src/appleseed-max-impl/oslutils.cpp

```diff
@@ -185,3 +191,33 @@ void connect_normal_map(
         normal_map_layer_name.c_str(), "TangentOut",
         material_node_name, material_tn_input_name);
 }
+
+void connect_sub_mtl(
+    asr::Assembly&          assembly,
+    asr::ShaderGroup&       shader_group,
+    const char*             shader_name,
+    const char*             shader_input,
+    Mtl*                    mat)
+{
+    auto compatible_mtl =   mat->ClassID() == AppleseedDisneyMtl::get_class_id() || 
+                            mat->ClassID() == AppleseedBlendMtl::get_class_id();
+
+    auto appleseed_mtl = static_cast<IAppleseedMtl*>(mat->GetInterface(IAppleseedMtl::interface_id()));
+    if (!appleseed_mtl || !compatible_mtl)
```

Comment:
I would be more confortable letting the caller do these tests and take appropriate action if they fail. The side benefit is that we would no longer need to include the Disney and Blend materials in this file.